src/digitalmarketplace/components/list-input/template.njk

```diff
@@ -0,0 +1,109 @@
+{%- from "../../../govuk-frontend/components/error-message/macro.njk" import govukErrorMessage -%}
+{%- from "../../../govuk-frontend/components/fieldset/macro.njk" import govukFieldset -%}
+{%- from "../../../govuk-frontend/components/hint/macro.njk" import govukHint -%}
+{%- from "../../../govuk-frontend/components/label/macro.njk" import govukLabel -%}
+{%- from "../../../govuk-frontend/components/input/macro.njk" import govukInput -%}
+
+{#-  We need this for error messages and hints as well -#}
+{%- set id = params.id -%}
+
+{#- A record of elements we want associated with the fieldset's aria-describedby -#}
+{%- set describedBy = params.fieldset.describedBy -%}
+
+{%- if params.hint %}
+  {%- set hintId = id + '-hint' -%}
+  {%- set describedBy = describedBy + ' ' + hintId if describedBy and hintId not in describedBy else hintId -%}
+{% endif -%}
+
+{%- if params.errorMessage %}
+  {%- set listErrorId = id + '-error' -%}
+  {%- set describedBy = describedBy + ' ' + listErrorId if describedBy and listErrorId not in describedBy else listErrorId -%}
+{% endif -%}
+
+<div class="dm-list-input{% if params.classes %} params.classes {% endif %}{% if params.errorMessage %} govuk-form-group--error{% endif %}">
+  {% if params.errorMessage %}
+    {{ govukErrorMessage({
+      id: listErrorId,
+      classes: 'dm-list-input-error-message' + params.errorMessage.classes if params.errorMessage.classes,
+      attributes: params.errorMessage.attributes,
+      html: params.errorMessage.html,
+      text: params.errorMessage.text,
+      visuallyHiddenText: params.errorMessage.visuallyHiddenText
+    }) | indent(2) | trim }}
+  {% endif %}
+  {% call govukFieldset({
+    describedBy: params.fieldset.describedBy if params.fieldset.describedBy else describedBy,
+    classes: params.fieldset.classes,
+    legend: {
+      html: params.fieldset.legend.text + ('<span class="dm-list-input__optional__optional"> (Optional)</span>' if params.optional) ,
```

Comment:
```suggestion
      html: params.fieldset.legend.text + ('<span class="dm-list-input__optional"> (Optional)</span>' if params.optional) ,
```

Reply:
I'm not sure whether having the optional text here makes the most sense; it might make more sense to define it closer to the point of use i.e. in the function that transforms the content loader question? What do you think?

Reply:
I think I agree, though it may depend on what approach we take to https://trello.com/c/YXsBOrDK/110-3-replace-text-boxes-with-govuk-frontend-text-input-component-in-create-a-dos-opportunity-journey

If we end up dropping questions as-is into templates on this repo, then this approach might work. If we transform things in content-loader or utils, then this isn't optimal.